kyuubi-main/src/test/scala/org/apache/kyuubi/engine/EngineSubmitSuite.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+import org.apache.kyuubi.{KyuubiSQLException, WithKyuubiServerWithMiniYarnService}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.operation.JDBCTestUtils
+
+class EngineSubmitSuite extends WithKyuubiServerWithMiniYarnService with JDBCTestUtils {
+  override protected def jdbcUrl: String = getJdbcUrl
+  override protected val kyuubiServerConf: KyuubiConf = KyuubiConf()
+  override protected val connectionConf: Map[String, String] = {
+    Map("spark.yarn.queue" -> "two_cores_queue",
+      "spark.master" -> "yarn",
+      "spark.executor.instances" -> "1",
+      "spark.driver.cores" -> "1",
+      "spark.executor.cores" -> "1",
```

Comment:
Can we simply request an executor or AM with exceeded cores to make the app hang?

Reply:
`spark.executor.instances` cann't be 0 so executor always exists.. If we want to use am to hold the resource, we need to tune other config to make app available without executor, otherwise app will wait until executor available.